persist/fs/commitlog/iterator.go

```diff
@@ -95,6 +95,7 @@ func (i *iterator) Next() bool {
 	var err error
 	i.read.series, i.read.datapoint, i.read.unit, i.read.annotation, err = i.reader.Read()
 	if err == io.EOF {
+		i.reader.Close()
 		// Try the next reader
 		i.reader = nil
```

Comment:
It could also encapsulate the `if i.reader != nil { i.reader.Close(); i.reader = nil }` logic as that's also duplicative.